### PR TITLE
Separating device CRUD notifications from physical device updates

### PR DESCRIPTION
**Pull request by giovannicuriel**
This PR separates the configuration used for:
 - Receiving messages from device manager (this PR sets it as always using kafka)
 - Sending device updates received from physical devices to other components (this is configurable - through kafka or through orion).

This is part of dojot/dojot#156

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -59,15 +56,21 @@ interface BrokerOptions {
...
   type?: "orion" | "kafka";
-
-  kafka?: KafkaOptions;
 }
 
 // Device manager options
 interface DeviceManagerOptions {
   // Device manager address.
   // This attribute includes port, such as localhost:4000.
   host: string;
```
Comment (ericbaum) [inline src/config.ts]: Is the device manager host address still required if the data is received via kafka?

File: src/kafka-handler.ts [hunk 2]
```diff
@@ -68,29 +58,75 @@ class KafkaHandler implements DataBroker {
...
+            console.log("Error: ", err);
+            console.log("Will try again in a few seconds.");
+            setTimeout(() => {
+              console.log("Trying again.");
+              this.initKafkaConfiguration(config, callback, "consumer");
+            }, 1000)
+        });
+
+        console.log("... Kafka consumer created and callbacks registered.");
+        this.initKafkaConfiguration(config, callback, "finish");
```
Comment (ericbaum) [inline src/kafka-handler.ts]: Why isn't this part a continuous sequence of code? Why the case 'finish': and recalling the function to reach that part of the code?

File: src/config.ts (hunk 1 above)
Comment (giovannicuriel) [reply src/config.ts]: No, it isn't.

File: src/kafka-handler.ts (hunk 2 above)
Comment (giovannicuriel) [reply src/kafka-handler.ts]: No need for the 'finish' case.

Comment (giovannicuriel) [reply src/kafka-handler.ts]: There is no need for *a* finish case. No need to create topics.

**Review by ericbaum — APPROVED** [review]

**Pull request merged**